test/com/google/javascript/jscomp/PeepholeRemoveDeadCodeTest.java

```diff
@@ -478,6 +481,163 @@ public void testOptimizeSwitch2() {
+        "    break;",
+        "  default:",
+        "    exit = 21;",
+        "    break;",
+        "}",
+        "switch(exit) {",
+        "  case 21: throw 'x';",
+        "  default : console.log('good');",
+        "}"), LINE_JOINER.join(
+        "var exit;",
+        "switch(exit) {",
+        "  case 21: throw 'x';",
+        "  default : console.log('good');",
+        "}"));
+  }
+
+  public void testOptimizeSwitchWithReturn() {
+    test(LINE_JOINER.join(
+        "function f() {",
+        "  switch('x') {",
+        "    case 'x': return 1;",
+        "    case 'y': return 2;",
+        "  }",
+        "}"),
+        "function f() { return 1; }");
+
+    // TODO(moz): This is to show that the current optimization might break if the input has block
+    // scoped declarations. We will need to fix the logic for removing blocks to account for this.
+    testEs6(LINE_JOINER.join(
+        "function f() {",
+        "  let x = 1;",
+        "  switch('x') {",
+        "    case 'x': { let x = 2; return 3; }",
+        "    case 'y': return 4;",
+        "  }",
+        "}"),
+        "function f() { let x = 1; let x = 2; return 3; }");
+  }
+
+  public void testOptimizeSwitchWithThrow() {
+    test(LINE_JOINER.join(
+        "function f() {",
+        "  switch('x') {",
+        "    case 'x': throw f;",
+        "    case 'y': throw f;",
+        "  }",
+        "}"),
+        "function f() { throw f; }");
+  }
+
+  public void testOptimizeSwitchWithContinue() {
+    test(LINE_JOINER.join(
+        "function f() {",
+        "  while (true) {",
+        "    switch('x') {",
+        "      case 'x': continue;",
+        "      case 'y': continue;",
+        "    }",
+        "  }",
+        "}"),
+        "function f() { while (true) { continue; } }");
+  }
+
+  // GitHub issue #1722: https://github.com/google/closure-compiler/issues/1722
+  public void testOptimizeSwitchWithDefaultCase() {
+    test(LINE_JOINER.join(
+        "function f() {",
+        "  switch('x') {",
+        "    case 'x': return 1;",
+        "    case 'y': return 2;",
+        "    default: return 3",
+        " }",
+        "}"),
+        "function f() { return 1; }");
+
+    test(LINE_JOINER.join(
+        "switch ('hasDefaultCase') {",
+        "  case 'foo':",
+        "    foo();",
+        "    break;",
+        "  default:",
+        "    bar();",
+        "    break;",
+        "}"),
+        "bar();");
+
+    test(LINE_JOINER.join(
+        "switch ('hasDefaultCase') {",
+        "  case 'foo':",
+        "    foo();",
+        "    break;",
+        "  default:",
+        "    foo();",
+        "    break;",
+        "    bar();",
+        "}"),
+        "foo();");
+
+    test(LINE_JOINER.join(
+        "switch (a()) {",
+        "  default:",
+        "    bar();",
+        "    break;",
+        "}"),
+        "a(); bar();");
+
+    test(LINE_JOINER.join(
+        "switch (a()) {",
+        "  default:",
+        "    break;",
```

Comment:
I think a more realistic example would be:
   default:
       if (condition) { break; }
       doStuff();